### ISSUE TITLE
Feature request: adding new parameters

### ISSUE THREAD

**Issue by doraskayo** [sibradzic/upp#21]
While the `set` option allows changing an existing table parameter, there doesn't seem to be an option to add new parameters to an existing PowerPlay table (or to remove them.)

In my case, the PowerPlay table in the VBIOS doesn't include the `EnableZeroRPM`/`FanStopTemperature`/`FanStartTemperature` parameters by the default, but the hardware does support them. In Windows, I assume (but haven't verified this) these are set via a "soft" PowerPlay table.

I would like to be able to add these parameters at runtime so this feature would also work on Linux.

**Comment by sibradzic** [comment]
First of all, what is your card?

I'm afraid your assumption is wrong. If PowerPlay table does not have some particular parameter, it probably means that such parameter was never meant to be there in the first place. Starting with the Radeon VII a.k.a. Vega 20, PowerPlay tables are featuring pretty strictly defined data structures, and the only flexibility it provides is enabling or disabling certain "feature flags" (flipping certain parameters between 0 and 1).

AFAIK, `EnableZeroRPM`, `FanStopTemperature` and `FanStartTemperature` are all Vega 56/64 only, they do not exist on other that those...

**Comment by doraskayo** [comment]
@sibradzic, my card is a NITRO+ RX 580 8GB (Polaris). It advertises a "ZERO DB COOLING" feature which is described as "Enjoy complete silence under light workloads with entirely passive cooling. No spinning fans means no distractions while you work or chill."

Also, the mentioned parameters are available at least for some Polaris cards as can be seen in this commit: https://gitlab.com/linux-kernel/linux/-/commit/9610a3bfde4100c67ecc3367ed34d43d918be88f

I had made the assumption that the "ZERO DB COOLING" feature is basically `EnableZeroRPM`. Perhaps I'm mistaken.

**Comment by sibradzic** [comment]
I've got the VBIOS from https://www.techpowerup.com/vgabios/191263/sapphire-rx580-8192-170320-1 (that matches your card?), and analysed the PP table in hex editor. It turned out that the last few values in the `Fan_Table` were not decoded at all, because the `Fan_Table` substructure auto-generated from Linux kernel sources didn't contain `ATOM_Polaris_Fan_Table` at all, so the parameters and values were decoded using old `ATOM_Tonga_Fan_Table` structure!

Now, after re-generating the data structures form the fresh kernel source, `FanStartTemperature`, `FanStopTemperature`, `EnableZeroRPM` as well as few other parameters are decoded correctly. Full list of added params in: https://github.com/sibradzic/upp/commit/8932873aca19463219ec9e9311dc6f035837f687#diff-3f3d61b3f79a7acd0c2bbb90081671cfe7de4a7dca483ef4fe0f963781d8b6ed

TLDR:
Thanks for reporting, this was a bug! Please upgrade to `0.1.0` via pip3.

**Comment by doraskayo** [comment]
Thanks @sibradzic!

The expected values are now listed:
```
FanTable:
  RevId: 9
  THyst: 3
  TMin: 4000
  TMed: 6500
  THigh: 8500
  PWMMin: 2000
  PWMMed: 4000
  PWMHigh: 6000
  TMax: 10900
  FanControlMode: 1
  FanPWMMax: 100 
  FanOutputSensitivity: 4836
  FanRPMMax: 2280
  MinFanSCLKAcousticLimit: 141100
  TargetTemperature: 75
  MinimumPWMLimit: 18
  FanGainEdge: 120 
  FanGainHotspot: 120 
  FanGainLiquid: 100 
  FanGainVrVddc: 120 
  FanGainVrMvdd: 120 
  FanGainPlx: 120 
  FanGainHbm: 100 
  EnableZeroRPM: 1
  FanStopTemperature: 46
  FanStartTemperature: 54
  Reserved: 0
```
